client/src/components/Lectures.tsx

```diff
@@ -1,15 +1,28 @@
-import React from "react";
+import React, { useState } from "react";
+import Pagination from "@material-ui/lab/Pagination";
 import { SectionWrapper } from "./Home";
 import Title from "./Title";
 import ScheduleGrid from "./ScheduleGrid";
+import { RootState } from "../reducers/index";
+import { connect } from "react-redux";
+import { generateSchedules, CourseSection } from "../util/testScheduler";
 
-function Lectures() {
+function Lectures({ schedules = [] }: any) {
+  const [selected, setSelected] = useState(0);
+  const handleChange = (e: any, n: number) => setSelected(n);
   return (
     <SectionWrapper>
       <Title title="4. Select Lectures to Lock Them"></Title>
-      <ScheduleGrid></ScheduleGrid>
+      {schedules[selected] && <ScheduleGrid schedule={schedules[selected]}/>}
+      <Pagination count={schedules.length} shape="rounded" onChange={handleChange} />
     </SectionWrapper>
   );
 }
 
-export default Lectures;
+const mapStateToProps = (state: RootState) => {
+  return {
+    schedules: generateSchedules(state.HomeReducer.sections),
```

Comment:
Random comment - interesting that we're calling generate schedules in the mapStateToProps.

Reply:
Found a [redux pattern](https://redux.js.org/recipes/computing-derived-data) called `selector` -- perhaps I can do that instead!

Reply:
Nice, yeah that would make it more efficient - but isn't too big a deal right now. Should we merge what we have first and maybe do this for polish if we have time?